lib/re2.rb

```diff
@@ -3,5 +3,12 @@
 #
 # Copyright (c) 2010-2014, Paul Mucur (http://mudge.name)
 # Released under the BSD Licence, please see LICENSE.txt
-require "re2.so"
+begin
+  ::RUBY_VERSION =~ /(\d+\.\d+)/
+  require_relative "#{Regexp.last_match(1)}/re2.so"
```

Comment:
I know this comes from Nokogiri but I wonder if it'd be better to avoid the global `Regexp.last_match` (even though it is thread-local):

```suggestion
  require_relative "#{RUBY_VERSION[/(\d+\.\d+)/, 1]}/re2.so"
```

Reply:
That works.